tools/pnnx/src/pass_ncnn/fuse_permute_gridsample.cpp

```diff
@@ -0,0 +1,125 @@
+// Tencent is pleased to support the open source community by making ncnn available.
+//
+// Copyright (C) 2023 THL A29 Limited, a Tencent company. All rights reserved.
+//
+// Licensed under the BSD 3-Clause License (the "License"); you may not use this file except
+// in compliance with the License. You may obtain a copy of the License at
+//
+// https://opensource.org/licenses/BSD-3-Clause
+//
+// Unless required by applicable law or agreed to in writing, software distributed
+// under the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR
+// CONDITIONS OF ANY KIND, either express or implied. See the License for the
+// specific language governing permissions and limitations under the License.
+
+#include "fuse_permute_gridsample.h"
+
+#include "pass_level2.h"
+
+#include <float.h>
+
+namespace pnnx {
+
+namespace ncnn {
+
+class fuse_permute_gridsample_4d_pass : public GraphRewriterPass
+{
+public:
+    const char* match_pattern_graph() const
+    {
+        return R"PNNXIR(7767517
+5 4
+pnnx.Input              input_a     0 1 a
+pnnx.Input              input_b     0 1 b
+Permute                 op_0        1 1 b b1 0=3 1=2
+GridSample              op_1        2 1 a b1 out 0=%c 1=%d 2=%e 3=0
+pnnx.Output             output      1 0 out
+)PNNXIR";
```

Comment:
move these fusion patterns into pass_ncnn/F_grid_sample.cpp
assign the pattern with higher priority
eg. `REGISTER_GLOBAL_PNNX_NCNN_GRAPH_REWRITER_PASS(F_grid_sample_1, 19)`
we could access input/output shape info there